### ISSUE TITLE
Synchronize model checkpointing and servable model saving

### ISSUE THREAD

**Issue by monatis** [qdrant/quaterion#29]
Currently, `TrainableModel.save_servable()` is called by the user at the end of the training loop. This is problematic because we may end up with saving an overfitted state of the model even if we are trying to monitor an evaluation metric with `pl.callbacks.ModelCheckpoint`. So we need to come up with a way to synchronize both.
## Possible solution
- We may need to subclass `ModelCheckpoint` inside `quaterion` for synchronization.
- We may accept additional keyword arguments in `Quaterion.fit` to automatically save a servable checkpoints to the specified directory with a specified interval.
